addons/hr_expense/models/hr_expense.py

```diff
@@ -152,14 +153,16 @@ def _compute_product_has_cost(self):
     @api.depends('sheet_id', 'sheet_id.account_move_id', 'sheet_id.state')
     def _compute_state(self):
         for expense in self:
-            if not expense.sheet_id or expense.sheet_id.state == 'draft':
+            if not expense.sheet_id:
                 expense.state = "draft"
+            elif expense.sheet_id.state == 'draft':
+                expense.state = "reported"
             elif expense.sheet_id.state == "cancel":
                 expense.state = "refused"
-            elif expense.sheet_id.state == "approve" or expense.sheet_id.state == "post":
+            elif expense.sheet_id.state in {"approve", "post"}:
```

Comment:
```suggestion
            elif expense.sheet_id.state in ('approve', 'post'):
```

Reply:
A set is faster (even with the generation, as seen with the rd-accounting team during a previous debate), not a huge boost in this case but why change it?

Reply:
@JulienAlardot Tested and that's true. You can ignore this part